### ISSUE TITLE
Compiling code with macros results in error

### ISSUE THREAD

**Issue by shadaj** [lihaoyi/scala-js-fiddle#10]
Compiling the last two examples from the homepage of http://scala-js-fiddle.com results in an error:
```
Main.scala:62: error: macro implementation not found: async
(the most common reason for that is that you cannot use macro implementations in the same compilation run that defines them)
    async{
         ^
```
and
```
Main.scala:293: error: macro implementation not found: foreach
(the most common reason for that is that you cannot use macro implementations in the same compilation run that defines them)
      for (x <- 0 until canvas.width optimized){
             ^
```

**Comment by sjrd** [comment]
Do you declare the right dependencies to the libraries providing these macros? What does your `libraryDependencies` look like? For the `async` stuff it should contain something like this:
```scala
libraryDependencies += "org.scala-lang.modules" %% "scala-async" % "0.9.2" % "provided"
```
This should work just fine out of the box.

If it doesn't, please provide a link to a branch in a GitHub repo exhibiting the error. Or at least the content of your `build.sbt` and `Main.scala`.

**Comment by lihaoyi** [comment]
Looks like this happened after my move from ec2 to heroku. I bet the
classloading used to make macros doesn't work on their filesystem layout or
something.

**Comment by shadaj** [comment]
I'm not running the server locally, the error is occurring on the hosted server. For example, http://www.scala-js-fiddle.com/gist/9443f8e0ecc68d1058ad/RayTracer.scala does not compile.

**Comment by lihaoyi** [comment]
Yeah I saw that. It's hosted on heroku now, after I moved it from EC2, and
I suspect that move is the cause of the problem
